presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -198,50 +203,67 @@ public ColumnarBinaryHiveRecordCursor(RecordReader<K, BytesRefArrayWritable> rec
             HiveColumnHandle column = columns.get(columnIndex);
             if (column.isPartitionKey()) {
                 HivePartitionKey partitionKey = partitionKeysByName.get(column.getName());
-                checkArgument(partitionKey != null, "Unknown partition key %s", column.getName());
+                checkArgument(partitionKey != null || isHiddenColumnHandle(column), "Unknown partition key %s", column.getName());
 
-                byte[] bytes = partitionKey.getValue().getBytes(UTF_8);
+                byte[] bytes;
+                String columnValue;
+                if (partitionKey != null) {
+                    columnValue = partitionKey.getValue();
+                    bytes = partitionKey.getValue().getBytes(UTF_8);
```

Comment:
This can be
```java
bytes = columnValue.getBytes(UTF_8);
```
At which point it can be moved outside the if/else (and thus the code can be moved to a utility class)